railties/lib/rails/generators/rails/app/app_generator.rb

```diff
@@ -163,6 +164,10 @@ def config_when_updating
           remove_file "config/initializers/permissions_policy.rb"
         end
       end
+
+      if uses_sprockets
+        insert_into_file "config/application.rb", %(require "sprockets/railtie"), after: /require\(["']rails\/all["']\)\n/
```

Comment:
`rails\/all` is not always present in this file. At this point, if `skip-activestorage` is passed for example it will not be there.

Reply:
Ah, another alternative is changing `sprockets-rails` to require the railtie, that way we don't need it in `application.rb` anymore.

Reply:
That would actually be way nicer. Since that’s how normal rails integration gems work. Will there be a backwards compatibility issue though?

Reply:
It would change a little bit when it is loaded, but that should not be a problem.

It would be required at this line https://github.com/rails/rails/blob/20db0845cddf6e90a82a782ef51196b9ecb3716d/railties/lib/rails/generators/rails/app/templates/config/application.rb.tt#L24.

Instead of this line https://github.com/rails/rails/blob/20db0845cddf6e90a82a782ef51196b9ecb3716d/railties/lib/rails/generators/rails/app/templates/config/application.rb.tt#L18

But, if an existing app still has the require, or is requiring `'rails/all'`, the second require would just return `false`, so no behavior would change.

Reply:
Ah nice. Let’s make it so.

Reply:
Ah, we already do it https://github.com/rails/sprockets-rails/blob/master/lib/sprockets/rails.rb#L3.

So all we need to do is to remove the requires and move to the gem to the Gemfile